Create SteeringSystem.md

mamollica42 opened the pull request.

JesseTNRoberts commented on Documentation/Signoffs/SteeringSystem.md:
```diff
@@ -0,0 +1,107 @@
+### Specifications & Constraints
+- Must measure angle of rotation of the steering wheel
+- Must use the Nissan Leaf OEM steering wheel as the input
+- Angle Shall be no more than 960 degrees to the left and right (2 and 2/3 rotations)
```
Shouldn't this be no less?

JesseTNRoberts commented on Documentation/Signoffs/SteeringSystem.md:
```diff
@@ -0,0 +1,107 @@
+embedded into the steering column of the vehicle. The sensor will measure the steering angle and send
+the data into a local microcontroller which will interpret the data and relay it to the simulation.
+
+### Specifications & Constraints
```
I think there are further constraints/specs that are appropriate. Like not loose position when power cycled?

JesseTNRoberts commented on Documentation/Signoffs/SteeringSystem.md:
```diff
@@ -0,0 +1,107 @@
+_Arduino UNO Pinout_
+
+
+### Analysis
```
The purpose of this section is to analyze your specific design to show that it will meet the constraints. Remove any additional unnecessary details that aren't directly related to your design, specs, and analysis. Convince me this will do what you say that it needs to by analyzing appropriately.

JesseTNRoberts replied:
This section is still bloated with unnecessary information. Prove your system will do what you said above and won't violate any constraints. Do it in an organized and explicit fashion.

JesseTNRoberts replied:
I see very little in the way of analysis here. Prove to me that it will work. That does not require paragraphs of writing. This section should be mostly numbers or simulation results. Include just enough writing so that what you are proving and how you proved it are clear.

mamollica42 commented:
The Steering Wheel System has been updated with the following changes:
1. Added Additional constraint regarding behavior when power is cycled.
2.  Added Wiring Schematic to show encoder connection to the board.
3.  Removed some unnecessary details in the analysis
4. Added some detail in the analysis that better describes how the constraints and specifications are being met

JesseTNRoberts commented on Documentation/Signoffs/SteeringSystem.md:
```diff
@@ -8,8 +8,10 @@ the data into a local microcontroller which will interpret the data and relay it
 - Must measure angle of rotation of the steering wheel
 - Must use the Nissan Leaf OEM steering wheel as the input
-- Angle Shall be no more than 960 degrees to the left and right (2 and 2/3 rotations)
+- Angle shall be no more than 960 degrees from the left boundary to the right boundary (2 and 2/3 rotations)
```
Again, shouldn't this be no less? Why are you requiring no more measurement than this?

mamollica42 replied:
The steering wheel is only able to turn 960 degrees total. If the measured angle was more than that, then we know the angle must be wrong. Therefore, the angle must be no more than 960 degrees from left to right.

JesseTNRoberts replied:
This does not logically track. The minimum rotation is 960 degrees. Why are you requiring no more than that? Further, you have not required that it have at least the minimum. 

If you needed to drive 200 miles and your car got 20 miles to the gallon, you need **at least** 10 gallons. However, it is not true that you must not have more than 10 gallons.

Your constraint is incorrect.

JesseTNRoberts commented on Documentation/Signoffs/SteeringSystem.md:
```diff
@@ -8,8 +8,10 @@ the data into a local microcontroller which will interpret the data and relay it
+- Must not lose steering position when power is cycled
+- System must be small enough to fit in the area of the vehicle without interfering with the driver or other subsytems
 
 The simulation requires an input for the steering angle of the vehicle. In the Nissan Leaf, the steering
```
This is not correct. There is no reason the measuring must be limited to this range.

JesseTNRoberts commented on Documentation/Signoffs/SteeringSystem.md:
```diff
@@ -40,11 +42,15 @@ _Rotary Encoder Pinout_
 
 _Arduino UNO Pinout_
 
+![image](https://user-images.githubusercontent.com/100802413/202334812-ddca2adf-fbd7-4da2-900d-f3958cb5c812.png)
```
This is not third party buildable. The signals being connected need to be enumerated and every connection shown. Also, I'm not sure how this picture was generated but it is not appropriate. Kicad, auto cad, etc are appropriate.

JesseTNRoberts commented on Documentation/Signoffs/SteeringSystem.md:
```diff
@@ -0,0 +1,115 @@
+accuracy constraint, the error of the angle sensed must be within 1% of the true angle to prevent false
+representation of the steering.
+
+### Schematic
```
The steering wheel mechanical system has no way to mount apparently. Not buildable.

JesseTNRoberts commented on Documentation/Signoffs/SteeringSystem.md:
```diff
@@ -0,0 +1,115 @@
+
+![image](https://user-images.githubusercontent.com/100802413/200971670-3beca395-ed8c-400a-b166-59c4ad6b9142.png)
+
+_2D-Model of Gear and Encoder Connection_
```
Encoder has nowhere to mount and does not have a way to connect to the gears. Not buildable.

JesseTNRoberts commented on Documentation/Signoffs/SteeringSystem.md:
```diff
@@ -0,0 +1,115 @@
+##### 2. Wiring Schematic
+
+
+![image](https://user-images.githubusercontent.com/100802413/200903742-825de5d3-3bc1-40d5-83c7-1e14c02fa940.png)
```
Do not include this. The needed information should be in your schematic.

JesseTNRoberts commented on Documentation/Signoffs/SteeringSystem.md:
```diff
@@ -0,0 +1,115 @@
+
+_Rotary Encoder Pinout_
+
+![image](https://user-images.githubusercontent.com/100802413/200881810-5902fe09-6d5b-472e-89ba-e14607246b68.png)
```
Do not include this. The needed information should be in your schematic.

JesseTNRoberts commented on Documentation/Signoffs/SteeringSystem.md:
```diff
@@ -0,0 +1,115 @@
+### Analysis
+##### 1. Objective
+
+The objective of this design is to measure the angle of the steering wheel in degrees. To achieve the objective is by install a Rotary encode near the steering column.
```
Get rid of this. Constraints and functionality aren't introduced here.

JesseTNRoberts commented on Documentation/Signoffs/SteeringSystem.md:
```diff
@@ -0,0 +1,115 @@
+
+##### 2. Rotary Encoder
+
+Position sensors such as rotary encoders will determine the rotating shafts. In response to rotational movement, it generates an electrical signal, either analog or digital.
```
This is not the purpose of this section.

JesseTNRoberts commented on Documentation/Signoffs/SteeringSystem.md:
```diff
@@ -86,30 +49,46 @@ where C is the number of counts, CPR is the Counts per Revolution, and A is the
+This steering wheel design will use a TRD-NA1024NW absolute rotary encoder with a resolution of 1024. The Encoder a voltage rating of 12 to 24 VDC, and a maximum current consumption of 70 mA. The TRD-NA1024NW has 10 data bits, a VDD pin, and a GND pin. This requires a microcontroller with at least 10 digital GPIO pins and that can supply the current of up to 70 mA to the rotary encoder.
 
-This steering wheel design will use a TRD-NA1024NW absolute rotary encoder with a resolution of 1024. The Encoder has a max frequency of 20 kHz, a voltage rating of 12 to 24 VDC, and a maximum current consumption of 70 mA. Given the maximum frequency of 20 kHz, the sampling rate must be at least 40 kHz to satify the Nyquist Theorem. Using equation 2 above, the degrees of accuracy for this encoder ...
+The encoder has 1024 unique 10-bit outputs than will be read into the MCU. The output of the encoder changes on rotation of the knob. The knob of the encoder will be connected to a gear system that is mounted to the steering column of the vehicle so that the steering wheel rotation will rotate the knob. Each 10-bit output will increment or decrement a count variable in the MCU code depending on t ...
```
How do you know that the micro and the encoder will communicate?

mamollica42 replied:
They will communicate because the output of the encoder is simply 10-bit Grey code that can be read by the MCU by the digital pins. Additionally, the current and voltage ratings are such that the encoder will not damage the MCU.

JesseTNRoberts replied:
That does not mean that they will communicate. Bits have voltage associated and different devices operate at different voltages. You haven't considered the output voltage of the encoder as compared to the acceptable voltage levels for the inputs.

Gray code is not a communication method. It is an encoding.

JesseTNRoberts commented on Documentation/Signoffs/SteeringSystem.md:
```diff
@@ -86,30 +49,46 @@ where C is the number of counts, CPR is the Counts per Revolution, and A is the
 Similarly, the accuracy of the angle is highly important to the selection of an encoder. To find the output sensitivy of the encoder, the following equation is used:
      
-     D = 360/(Resolution * 2)       (2)
+     D = 360/(Resolution)       (2)
```
The resolution was not analyzed. I saw the Matlab script but it doesn't prove that the resolution is sufficient for your purpose. Probably that is because you may not have a firm idea of what the required resolution is. What resolution is needed for the application?

JesseTNRoberts commented on Documentation/Signoffs/SteeringSystem.md:
```diff
@@ -86,30 +49,46 @@ where C is the number of counts, CPR is the Counts per Revolution, and A is the
-##### 5. Microcontroller Selection
+##### 3. Microcontroller Selection
 
 The Arduino UNO REV3 is clocked at 16 MHz, has 14 digital I/O pins, and 6 analog I/O pins. The board can be powered via battery, USB, or the VDD pin and can operate within a voltage range of 7-12 V. Each GPIO pin can supply up to 40 mA, but normally operates at 20 mA per pin. Given that the encoder requires 10 digital pins, the Arduino UNO can support that requirement and still have leftover pins ...
```
This does not sound like all the electrical considerations have been taken. Why is the current supplied by the pins of concern? The encoder will output to the microcontroller, not the other way around. Further, what about voltage?

mamollica42 replied:
You are correct, I had it backwards when I first wrote it. I fixed it in the document. The Arduino pins can receive up to 40 mA and operate at 5 V. The Encoder outputs a maximum of 32 mA and the pins output 10-bit greycode.

JesseTNRoberts replied:
Again grey code is not a method of communication. It is an encoding. Nothing (in the signoff) says that these will be able to communicate when you plug them up. "bits" don't exist electrically. They become voltages (usually). Different devices use different voltages to represent a 1 vs a 0. A "1" in some systems is any voltage greater than 1.5 v while in others it could be anything greater than 14v.

JesseTNRoberts commented on Documentation/Signoffs/SteeringSystem.md:
```diff
@@ -0,0 +1,126 @@
+- Must use the Nissan Leaf OEM steering wheel as the input
+- Angle shall be no more than 960 degrees from the left boundary to the right boundary (2 and 2/3 rotations)
+- Error of the angle must be within 1% of the true steering angle
+- Must not lose steering position when power is cycled
```
Not proven in the analysis section.

JesseTNRoberts commented on Documentation/Signoffs/SteeringSystem.md:
```diff
@@ -0,0 +1,126 @@
+- Angle shall be no more than 960 degrees from the left boundary to the right boundary (2 and 2/3 rotations)
+- Error of the angle must be within 1% of the true steering angle
+- Must not lose steering position when power is cycled
+- System must be small enough to fit in the area of the vehicle without interfering with the driver or other subsytems
```
Not analyzed.

JesseTNRoberts commented on Documentation/Signoffs/SteeringSystem.md:
```diff
@@ -0,0 +1,126 @@
+is limited to 2 and 2/3 total rotations from the clockwise boundary to the counter-clockwise boundary.
+Therefore, the angle measured must be limited to 960 degrees total. Due to the broader impact 
+accuracy constraint, the error of the angle sensed must be within 1% of the true angle to prevent false
+representation of the steering.
```
Within 1% suggests that you will have a precision of 0.1 degrees. However, you then later say the target precision is 9.6 degrees. That is very imprecise. 0.1 degrees seems an appropriate target.

JesseTNRoberts commented on Documentation/Signoffs/SteeringSystem.md:
```diff
@@ -8,15 +8,15 @@ the data into a local microcontroller which will interpret the data and relay it
-- Angle shall be no more than 960 degrees from the left boundary to the right boundary (2 and 2/3 rotations)
-- Error of the angle must be within 1% of the true steering angle
+- Angle shall be no less than 960 degrees from the left boundary to the right boundary (2 and 2/3 rotations)
+- Error of the angle must be within 1 degree of the true steering angle
```
My follow up question is why? Where does this constraint come from and why is 1 degree an appropriate constraint?

JesseTNRoberts commented on Documentation/Signoffs/SteeringSystem.md:
```diff
@@ -0,0 +1,152 @@
+
+The furthest point that the design extends to the right is the small gear on the steering column, and its farthest point protrudes 2.03 cm from edge of the steering column. The furthest point to the left is the large gear and it should extend 10.38 cm past the edge of the mounting plate. The model should not extend below the mounting plate by any more than 1 cm. The back of the mounting plate is  ...
+
+### BOM
```
This design looks good. I'm ready to signoff. But, I am a bit concerned about the encoder selection. It is somewhat expensive. I recommend looking for a lower cost option (either a lower cost encoder or maybe a rotary potentiometer). However, I'll leave that as a decision for the team. Keep in mind that the total budget is 1000 dollars without specifically showing a unique need for additional funds. So, be careful to consider if the money may be needed somewhere else in the project.

JesseTNRoberts commented:
I think I need to be added to have write access. I can't merge the pull request.
